backend/cfg/cfg.mli

```diff
@@ -31,9 +31,55 @@ include module type of struct
   include Cfg_intf.S
 end
 
+module BasicInstructionList : sig
+  type instr = basic instruction
+
+  type cell
+
+  val insert_before : cell -> instr -> unit
+
+  val insert_after : cell -> instr -> unit
+
+  val instr : cell -> instr
```

Comment:
It may be possible to hide `cell` entirely, it looks like all uses go through iter_cell, so `~f`  could return a variable 
`type operation = NOP | INSERT_BEFORE of instr | INSERT_AFTER of instr`
although the meaning of iter with INSERT_AFTER needs to be clarified (will the new element be included in the iteration?).

Reply:
I am not sure about that one: the representation
is very much tied to the representation of the
list. It is arguably not _that_ true with the current
dummy implementation, but should be more
apparent with a proper implementation.

Reply:
We discussed it offline and it will be addressed in a follow-up PR.